### PR TITLE
Allow `[x/y/z]scale = Base.Fix1(log, base::Number)`

### PR DESCRIPTION
**Pull request by asinghvi17**
simply hooks up `Base.Fix1{typeof(log), <: Number}` to the Axis transform api.

# Description

- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] Added an entry in NEWS.md (for new features and breaking changes)
- [ ] Added or changed relevant sections in the documentation
- [ ] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

File: src/makielayout/blocks/axis.jl [hunk 1]
```diff
@@ -1366,7 +1366,7 @@ defaultlimits(::typeof(Makie.pseudolog10)) = (0.0, 100.0)
 defaultlimits(::Makie.Symlog10) = (0.0, 100.0)
 
 defined_interval(::typeof(identity)) = OpenInterval(-Inf, Inf)
-defined_interval(::Union{typeof(log2), typeof(log10), typeof(log)}) = OpenInterval(0.0, Inf)
+defined_interval(::Union{typeof(log2), typeof(log10), typeof(log), Base.Fix1{typeof(log), <: Number}) = OpenInterval(0.0, Inf)
```
Comment (asinghvi17) [inline src/makielayout/blocks/axis.jl:1369]:
```suggestion
defined_interval(::Union{typeof(log2), typeof(log10), typeof(log), Base.Fix1{typeof(log), <: Number}}) = OpenInterval(0.0, Inf)

**Comment by SimonDanisch** [comment]
For what is this exactly?

**Comment by asinghvi17** [comment]
Log scale by arbitrary base.  I just have to define the tick formatters now...

**Comment by SimonDanisch** [comment]
Closing this for now, since the use case isn't 100% clear.

**Pull request closed without merging**